internal/engine/integration/procedure_test.go

```diff
@@ -256,6 +256,23 @@ func Test_Procedures(t *testing.T) {
 			}`,
 			outputs: [][]any{{int64(1), int64(3)}},
 		},
+		{
+			name: "decimal",
+			procedure: `procedure d() public view {
+				$i := 100.423;
+				$j decimal(16,8) := 46728954.23743892;
+				$k := $i::decimal(16,8) + $j;
+				if $k != 46729054.66043892 {
+					error('decimal failed');
+				}
+				if $k::text != '46729054.66043892' {
+					error('decimal text failed');
+				}
+				if ($k::decimal(16,2))::text != '46729054.66' {
```

Comment:
A dumb question, just curious, cut the scale won't round the value right? It's a silly question as I typed it out

Reply:
```psql://postgres@[local]:5432/postgres 
# select 3.1415926::numeric(10,3);
┌─────────┐
│ numeric │
├─────────┤
│   3.142 │
└─────────┘
(1 row)

Time: 0.506 ms

```

That behavior?

Reply:
ah so it does round the number?

Reply:
I'm not sure the behavior, just want to clearfy what's the behavior

Reply:
> If the scale of a value to be stored is greater than the declared scale of the column, the system will round the value to the specified number of fractional digits. Then, if the number of digits to the left of the decimal point exceeds the declared precision minus the declared scale, an error is raised. For example, a column declared as
> 
> NUMERIC(3, 1)
> 
> will round values to 1 decimal place and can store values between -99.9 and 99.9, inclusive.

https://www.postgresql.org/docs/current/datatype-numeric.html

Reply:
The error case described looks like this:

```
# select 1234.6::numeric(5,2);
ERROR:  numeric field overflow
DETAIL:  A field with precision 5, scale 2 must round to an absolute value less than 10^3.
Time: 10.250 ms
```


5 - 2 = 3, but that requires 4 digits for 1234, so it errors.

Not sure if that's an issue for us, or just a predictable runtime error.

Reply:
I think that is the runtime error we want